viewing_party/party.py

```diff
@@ -1,23 +1,177 @@
+
+    index = 0
+    genres = []
+    least_popular = []
+    most_popular = []
+    
+    if user_data == {"watched":[]}:
+        return None
+    else:
+        for movie in user_data["watched"]:
+            index +=1
+            genres.append(user_data["watched"][index-1]["genre"])
+        for item in genres:
+            if item not in least_popular:
+                least_popular.append(item)
+            else:
+                most_popular.append(item)
+        return (most_popular[0])
 
-# -----------------------------------------
 # ------------- WAVE 3 --------------------
-# -----------------------------------------
 
-        
-# -----------------------------------------
-# ------------- WAVE 4 --------------------
-# -----------------------------------------
+def get_unique_watched(user_data):
+    unique = []
+    not_unique = []
+    total_movies_list = []
+    users_movies = []
+    friends_movies = []
+    for movie in user_data["watched"]:
+        users_movies.append(movie)
+    for movie in user_data["friends"]:
+        for item in movie["watched"]:
+            friends_movies.append(item)
+    total_movies_list = users_movies + friends_movies
+    
+    for item in total_movies_list:
+        if item not in friends_movies:
+            unique.append(item)
+        else:
+            not_unique.append(item)
+    return unique
+
+def get_friends_unique_watched(user_data):
+    unique = []
+    users_movies = []
+    friends_movies = []
+    for movie in user_data["watched"]:
+        users_movies.append(movie)
+
+    for movie in user_data["friends"]:
+        for item in movie["watched"]:
+            friends_movies.append(item)
+            
+    for item in friends_movies:
+        if item not in users_movies:
+            if item not in unique:
+                unique.append(item)
+    
+    return unique
+# 
+# ------------- WAVE 4 -------------
+
+def get_available_recs(user_data):
+    unique = []
+    users_movies = []
+    friends_movies = []
+    for movie in user_data["watched"]:
+        users_movies.append(movie)
+
+    for movie in user_data["friends"]:
+        for item in movie["watched"]:
+            friends_movies.append(item)
+            
+    for item in friends_movies:
+        if item not in users_movies:
+            if item not in unique:
+                if item["host"] in user_data["subscriptions"]:
+                    unique.append(item)
+    
+    return unique
 
-# -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+    most_genre = []
+    unique = []
+    users_movies = []
+    friends_movies = []
+    
+    for movie in user_data["watched"]:
+        users_movies.append(movie)
+        most_genre.append(movie["genre"])
+
+    if len(users_movies) == 0:
+        return friends_movies
+    the_max = max(set(most_genre), key= most_genre.count)

```

Comment:
It looks like `the_max` is not being used anywhere in the function. What was the purpose of this variable?